### PR TITLE
feat: add tag-and-release workflow

### PR DESCRIPTION
**Pull request by jackhodgkiss**

### CODE REVIEW

File: .github/workflows/tag-and-release.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          git remote add origin "https://github.com/openstack/$(basename $(pwd)).git"
+      - name: Github checkout 🛎 [${{ env.naming-things-is-hard-repository }}]
+        uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846
+        with:
+          repository: ${{ env.naming-things-is-hard-repository }}
+          path: naming-things-is-hard
+      - name: Setup Python 🐍
+        uses: actions/setup-python@98f2ad02fd48d057ee3b4d4f66525b231c3e52b6
+        with:
+          python-version: '3.x'
```
Comment (markgoddard) [inline .github/workflows/tag-and-release.yml:32]: As discussed - this step is probably not required here.

File: .github/workflows/tag-and-release.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      naming-things-is-hard-repository: stackhpc/naming-things-is-hard
+    name: Automatic tagging for ${{ github.ref_name }} 🏷
+    steps:
+      - name: Github checkout 🛎 [${{ github.repository }}]
+        uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846
+        with:
+          fetch-depth: 0
+      - name: Configure repository 🛠
+        run: |
+          git remote rename origin stackhpc
```
Comment (markgoddard) [inline .github/workflows/tag-and-release.yml:22]: When running multiple shell commands, it's good practice to use `&&` to ensure that the whole step fails if the first command fails.

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: .github/workflows/tag-and-release.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    name: Automatic tagging for ${{ github.ref_name }} 🏷
+    steps:
+      - name: Github checkout 🛎 [${{ github.repository }}]
+        uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846
+        with:
+          fetch-depth: 0
+      - name: Configure repository 🛠
+        run: |
+          git remote rename origin stackhpc &&
+          git remote add origin "https://github.com/openstack/$(basename $(pwd)).git"
```
Comment (markgoddard) [inline .github/workflows/tag-and-release.yml:23]: Need to push all the tags to the stackhpc remote.

File: .github/workflows/tag-and-release.yml [hunk 4]
```diff
@@ -20,7 +20,9 @@ jobs:
       - name: Configure repository 🛠
         run: |
           git remote rename origin stackhpc &&
-          git remote add origin "https://github.com/openstack/$(basename $(pwd)).git"
+          git remote add origin "https://github.com/openstack/$(basename $(pwd)).git" &&
+          git fetch origin &&
+          git push --follow-tags
```
Comment (jackhodgkiss) [inline .github/workflows/tag-and-release.yml:25]: Tags from upstream are fetched and then pushed into the downstream repository.

File: .github/workflows/tag-and-release.yml (hunk 3 above)
Comment (jackhodgkiss) [reply .github/workflows/tag-and-release.yml]: This has now been addressed in the latest commit.

File: .github/workflows/tag-and-release.yml (hunk 4 above)
Comment (markgoddard) [inline .github/workflows/tag-and-release.yml:25]: I think we need to be specific about which remote we are pushing to.

Comment (jackhodgkiss) [reply .github/workflows/tag-and-release.yml]: Yeah makes sense to be clear with regards to where the push is targeting. This has been addressed in the latest commit.

File: .github/workflows/tag-and-release.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
+---
+name: Tag & Release
+'on':
+  push:
```
Comment (markgoddard) [inline .github/workflows/tag-and-release.yml:4]: Should we do the usual workflow_call pattern here?

File: .github/workflows/tag-and-release.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Github checkout 🛎 [${{ github.repository }}]
+        uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846
+        with:
+          fetch-depth: 0
+      - name: Configure repository 🛠
+        run: |
+          git remote rename origin stackhpc &&
+          git remote add origin "https://github.com/openstack/$(basename $(pwd)).git" &&
+          git fetch origin &&
+          git push stackhpc --follow-tags
```
[the remark this answers is not in the document]
Comment (jackhodgkiss) [reply .github/workflows/tag-and-release.yml:28]: Those commands have now been moved into their own step.

File: .github/workflows/tag-and-release.yml (hunk 5 above)
Comment (jackhodgkiss) [reply .github/workflows/tag-and-release.yml:4]: Good catch. Otherwise we can't use it as a reusable workflow. This has been fixed in 7c562e4.

Comment (markgoddard) [reply .github/workflows/tag-and-release.yml:4]: Do we want the push event in this one? Usually the reusable workflows just have workflow_call

**Review by markgoddard — APPROVED** [review]

**Pull request merged**